specs/Standards/ContractMetadata.md

```diff
@@ -0,0 +1,43 @@
+# Contract Metadata
+
+## [NEP-330](https://github.com/near/NEPs/discussions/329)
+
+Version `1.0.0`
+
+## Summary
+
+The contract metadata is a standard interface that allows auditing and viewing source code for a deployed smart contract. Implementation of this standard is purely optional but is recommended for developers whose contracts are open source.
+
+## Motivation
+
+There is no trivial way of finding the source code or author of a deployed smart contract. Having a standard that outlines how to view the source code of an arbitrary smart contract creates an environment of openness and collaboration.
+
+The initial discussion can be found [here](https://github.com/near/NEPs/discussions/329).
+
+## Interface
+
+Metadata applies at the contract level (`ContractMetadata`):
+
+```ts
+type ContractMetadata = {
+  version: string|null, // optional, commit hash being used for the currently deployed wasm. If the contract is not open-sourced, this could also be a numbering system for internal organization / tracking such as "1.0.0" and "2.1.0".
+  link: string|null, //optional,  link to open source code such as a Github repository or a CID to somewhere on IPFS.
```

Comment:
@BenKurrek @austinabell @MaximusHaximus what are your thoughts of adding a system field so we can get some info on what system helped generate the wasm as well.  This way we can try to rebuild the wasm and compare the hashes for auditing purposes.

Reply:
`system: "local"|string,`

Reply:
I mean, you would need architecture and toolchain version at least. Would the purpose of this be human-readable so someone could try to replicate it? You would also need some commit or version since the link doesn't specify repo links to commit that it was built with

Reply:
yeah ideally i want to be able to reference the source code and source dependencies to help with auditing the contract.  Here's what polygon does which is pretty cool: https://polygonscan.com/address/0x217cF04C783818E5b15Ae0723b22Ee2415Ab5fe3#code

Reply:
I was assuming that the toolchain version would be kept in the Github repo.  I don't really see the need to add a system field but would love other people's opinions